src/Subscriber/ConfigurationSubscriber.php

```diff
@@ -17,11 +18,20 @@ class ConfigurationSubscriber implements EventSubscriberInterface
 
     private array $communicationParameters;
 
-    public function __construct(Communication $config, array $fieldRoles, array $communicationParameters)
+    private array $addParams;
+
+    public function __construct(
+        Communication $config,
+        array $fieldRoles,
+        array $communicationParameters,
+        array $initialConfigurationParams = [],
+        array $initialNavigationParams = []
```

Comment:
Since we are binding constructor argument by the name $initialNavigationParams, placing the argument with same name here does not have much sense. We don't want to pass the same parameters to the search page and category page and i think that this would happen if it stay like this.